service/storage_proxy.cc

```diff
@@ -1765,6 +1765,8 @@ inline std::ostream& operator<<(std::ostream& os, const hint_wrapper& h) {
 
 using namespace std::literals::chrono_literals;
 
+static thread_local const utils::updateable_value_source<db::hints::host_filter> allow_all_hint_filter{db::hints::host_filter(db::hints::host_filter::enabled_for_all_tag())};
```

Comment:
Can't we store it outside a global? in storage_proxy maybe?

Reply:
The name is supper confusing too. 
See my comment at the end of this patch.

Reply:
Done - I made a separate constructor overload for `manager` that takes a non-updateable `host_filter` value.